### PR TITLE
use decrypt flag

### PR DESCRIPTION
**Pull request by alexanderjophus**
#### What this PR does / why we need it:

Plugs in the decrypt flag

#### Which issue(s) this PR fixes:

Fixes #63231

## Steps to reproduce

Create a config with a password, set it to hidden

Currently it's decrypted regardless of if the flag is passed in

#### Does this PR introduce a user-facing change?
```release-note
When using kots get config, now decrypts only if the --decrypt flag has been passed.
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: cmd/kots/cli/get-config.go [hunk 1]
```diff
@@ -209,18 +210,24 @@ func decryptGroups(clientset kubernetes.Interface, namespace string, groups []v1
...
 		for idx, item := range group.Items {
-			if item.Type == "password" {
+			if item.Type == "password" && decrypt {
 				// attempt to decrypt the password's value and default
-				decrypted, err := decryptString(item.Value.String())
-				if err == nil {
+				if item.Value.String() != "" {
+					decrypted, err := decryptString(item.Value.String())
+					if err != nil {
+						continue // don't fail if we can't decrypt
```
Comment (alexanderjophus) [inline cmd/kots/cli/get-config.go:218]: Not sure we want continue here. If `value` fails, we still probably want to decode `default`?

File: cmd/kots/cli/get-config.go [hunk 2]
```diff
@@ -234,6 +222,21 @@ func decryptGroups(clientset kubernetes.Interface, namespace string, groups []v1
...
 }
 
+func extractString(item string) multitype.BoolOrString {
+	if item == "" {
+		return multitype.BoolOrString{}
+	}
+	decrypted, err := decryptString(item)
+	if err != nil {
+		fmt.Println(err)
+		return multitype.BoolOrString{} // don't fail if we can't decrypt
```
Comment (sgalsaleh) [inline cmd/kots/cli/get-config.go:232]: why not fail?

**Review by cbodonnell — APPROVED** [review]

File: cmd/kots/cli/get-config.go [hunk 3]
```diff
@@ -129,7 +130,7 @@ func getConfigCmd(cmd *cobra.Command, args []string) error {
 		return errors.Wrap(err, "failed to get config")
 	}
 
-	config.ConfigGroups, err = decryptGroups(clientset, namespace, config.ConfigGroups)
+	config.ConfigGroups, err = decryptGroups(clientset, namespace, config.ConfigGroups, decrypt)
```
Comment (sgalsaleh) [inline cmd/kots/cli/get-config.go:133]: why not just skip this whole function if `decrypt` is false?

File: cmd/kots/cli/get-config.go (hunk 2 above)
Comment (alexanderjophus) [reply cmd/kots/cli/get-config.go:232]: Could do, I just wasn't 100% sure what may break, currently we don't do anything with the error

Comment (sgalsaleh) [reply cmd/kots/cli/get-config.go:233]: Let's fail here please so that customers aren't confused if their passwords were empty

Comment (sgalsaleh) [reply cmd/kots/cli/get-config.go:233]: Ah I see that we weren't checking the error before, so this is fine

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**